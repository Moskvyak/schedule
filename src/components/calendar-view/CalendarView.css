.calendar-view__body {
  display: flex;
  max-height: 600px;
  overflow: scroll;
  position: relative;
  padding-left: 75px;
}

.calendar-view__card {
  position: relative;
  width: 200px;
  margin: 0 20px;
  padding: 0 20px 20px 20px;
  border-left: 1px solid #888;
  border-right: 1px solid #888;
  background: rgba(255, 255, 255, 0.75);
  z-index: 1;
}
