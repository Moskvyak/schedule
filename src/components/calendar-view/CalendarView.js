import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types'
import moment from 'moment';

import { CalendarCard } from '../calendar-card/CalendarCard';
import { CalendarHeader } from '../calendar-header/CalendarHeader';
import { CalendarTimeGrid } from '../calendar-time-grid/CalendarTimeGrid';

import './CalendarView.css';

export class CalendarView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rowHeight: 80
    };
  }

  increaseRowHeight = () => {
    this.setState(prevState => ({ rowHeight: prevState.rowHeight + 10 }));
  };

  decreaseRowHeight = () => {
    if (this.state.rowHeight > 50) {
      this.setState(prevState => ({ rowHeight: prevState.rowHeight - 10 }));
    }
  };

  getNumberOfHours = () => {
    const { start, end } = this.props;
    return moment(end).diff(moment(start), 'hours');
  };

  render() {
    const { people, start } = this.props;
    const { rowHeight } = this.state;
    const numberOfRows = this.getNumberOfHours();
    return (
      <Fragment>
        <CalendarHeader
          people={people}
          increaseRowHeight={this.increaseRowHeight}
          decreaseRowHeight={this.decreaseRowHeight}
        />
        <div
          className="calendar-view__body"
          style={{ height: rowHeight * numberOfRows }}
        >
          <CalendarTimeGrid
            start={start}
            rowHeight={rowHeight}
            numberOfRows={numberOfRows}
          />
          {people.map(person =>
            <div
              className="calendar-view__card"
              key={person.id}
              style={{ height: rowHeight * numberOfRows }}
            >
              {person.cards.map(card =>
                <CalendarCard
                  {...card}
                  rangeStart={start}
                  key={`${card.id}-${card.type}`}
                  rowHeight={rowHeight}
                />
              )}
            </div>
          )}
        </div>
      </Fragment>
    );
  }
}

CalendarView.propTypes = {
    start: PropTypes.instanceOf(Date).isRequired,
    end: PropTypes.instanceOf(Date).isRequired,
    people: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        card: PropTypes.array
    }))
}
