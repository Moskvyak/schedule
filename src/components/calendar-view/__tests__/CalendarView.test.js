import React from 'react';
import renderer from 'react-test-renderer';
import { CalendarView } from '../CalendarView';

describe('<CalendarView />', () => {
  const requiredProps = {
    people: [
      {
        id: '1',
        name: 'test',
        cards: []
      },
      {
        id: '2',
        name: 'test2',
        cards: []
      }
    ],
    start: new Date('2018-09-01T00:00:00Z'),
    end: new Date('2018-09-01T24:00:00Z')
  };

  it('should render with required props', () => {
    const component = renderer.create(<CalendarView {...requiredProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
