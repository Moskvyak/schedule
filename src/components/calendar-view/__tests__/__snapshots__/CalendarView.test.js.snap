// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarView /> should render with required props 1`] = `
Array [
  <div
    className="calendar-header__wrapper"
>
    <div
        className="calendar-header__timelane"
    >
        <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus fa-w-14 fa-lg "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
        >
            <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
            />
        </svg>
        <svg
            aria-hidden="true"
            className="svg-inline--fa fa-minus fa-w-14 fa-lg "
            data-icon="minus"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
        >
            <path
                d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
            />
        </svg>
    </div>
    <div
        className="calendar-header__card"
    >
        <h3 />
    </div>
    <div
        className="calendar-header__card"
    >
        <h3 />
    </div>
</div>,
  <div
    className="calendar-view__body"
    style={
        Object {
            "height": 1920,
          }
    }
>
    <div
        className="calendar-time-grid__wrapper"
        style={
            Object {
                "height": 1920,
              }
        }
    >
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            12:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            1:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            2:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            3:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            4:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            5:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            6:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            7:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            8:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            9:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            10:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            11:00 AM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            12:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            1:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            2:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            3:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            4:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            5:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            6:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            7:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            8:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            9:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            10:00 PM
        </div>
        <div
            className="calendar-time-grid__row"
            style={
                Object {
                    "height": 80,
                    "lineHeight": "79px",
                  }
            }
        >
            11:00 PM
        </div>
    </div>
    <div
        className="calendar-view__card"
        style={
            Object {
                "height": 1920,
              }
        }
    />
    <div
        className="calendar-view__card"
        style={
            Object {
                "height": 1920,
              }
        }
    />
</div>,
]
`;
